lib/IRGen/GenDistributed.cpp

```diff
@@ -269,9 +269,17 @@ static CanSILFunctionType getAccessorType(IRGenModule &IGM,
     auto *actor = getDistributedActorOf(Target);
     assert(actor);
 
-    for (auto *genericParam : actor->getInnermostGenericParamTypes())
+    for (auto *genericParam : actor->getInnermostGenericParamTypes()) {
       genericParams.push_back(genericParam);
 
+      // and also forward all requirements this generic parameter might have.
+      for (auto req : actor->getGenericRequirements()) {
+        if (req.getFirstType()->isEqual(genericParam)) {
+          genericRequirements.push_back(req);
```

Comment:
the fix™

Reply:
Why not just add all requirements from the actor's generic signature, or is there a specific reason you're doing this? What you have here is not correct for a few reasons:
- req.getFirstType() might be a member type (`T.Element: Hashable`)
- it's not sufficient to just collect the requirements that immediately mention the type (you might have `T == U`, `U: P`)

Reply:
I don't know what the expected thing to do is, so if adding all is the expected thing let's do that! Thanks for the snippet, I'll give that a go

Reply:
I feel like very similar thing came up before but we couldn't make it work, maybe this time we can pull it off!

Reply:
Yeah I'm definitely struggling to have the version with `buildGenericSignature` not crash either in parameter count verification or at runtime... Trying still though.

Reply:
Okey I understand finally -- we need to have the param at the depth, and we need to detect that from the outer actor's scope which depth we're at